cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java

```diff
@@ -331,14 +338,17 @@ private String getAggregation(MetricDataQuery query) {
 
   @Override
   public void deleteBefore(long timestamp) throws Exception {
-    // Delete all data before the timestamp. null for MeasureName indicates match any MeasureName.
-    for (int resolution : resolutions) {
-      // NOTE: we do not purge on TTL the "totals" currently, as there might be system components dependent on it
-      if (TOTALS_RESOLUTION == resolution) {
-        continue;
-      }
-      CubeDeleteQuery query = new CubeDeleteQuery(0, timestamp, resolution, Maps.<String, String>newHashMap());
-      cube.get().delete(query);
+    for (int resolution : resolutionTTLMap.keySet()) {
+      // Delete all data before the timestamp. null for MeasureName indicates match any MeasureName.
+      deleteMetricsBeforeTimestamp(timestamp, resolution);
+    }
+  }
+
+  @Override
+  public void deleteTTLExpired() throws Exception {
+    long currentTime = TimeUnit.SECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS);
```

Comment:
`TimeUnit.SECONDS.toMillis(System.currentTimeMillis());`

Reply:
Changed to `TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());` since we want seconds here